qiskit_machine_learning/datasets/ad_hoc.py

```diff
@@ -31,240 +31,618 @@ def ad_hoc_data(
+    n : int
+        Number of qubits (dimension of the feature space).
+    gap : int, optional
+        Separation gap :math:`\Delta` when ``labelling_method="expectation"``.
+        Default is 0.
+    plot_data : bool, optional
+        If True, plots the sampled data (disabled automatically if ``n > 3``).
+        Default is False.
+    one_hot : bool, optional
+        If True, returns labels in one-hot format. Default is True.
+    include_sample_total : bool, optional
+        If True, the function also returns the total number of accepted samples
+        (those that fall outside the ``\pm \Delta`` zone). Default is False.
+    entanglement : str, optional
+        Determines which second-order terms :math:`Z_i Z_j` appear in
+        :math:`U_{\Phi(\vec{x})}`. The options are:
+
+        * ``"linear"``: Includes terms :math:`Z_i Z_{i+1}`.
+        * ``"circular"``: Includes ``"linear"`` terms plus :math:`Z_{n-1}Z_0`.
+        * ``"full"``: Includes all pairwise terms :math:`Z_i Z_j`.
+
+        Default is ``"full"``.
+    sampling_method : str, optional
+        The method used to generate uniform samples :math:`\vec{x}`:
+
+        * ``"grid"``: Generates a uniform grid and selects points (supported only
+          if ``n <= 3``).
+        * ``"hypercube"``: Uses a variant of Latin hypercube sampling for
+          stratified samples.
+        * ``"sobol"``: Uses Sobol sequences.
+
+        Default is ``"grid"``.
+    divisions : int, optional
+        If ``sampling_method="hypercube"``, this parameter determines the number
+        of stratifications along each dimension. Should be chosen close to
+        ``training_size``. Default is 0 (unused by other methods).
+    labelling_method : str, optional
+        Method for assigning labels:
+
+        * ``"expectation"``: Uses the expectation value of :math:`O`.
+        * ``"measurement"``: Performs a measurement in the computational basis.
+
+        Default is ``"expectation"``.
+    class_labels : list, optional
+        Custom labels for the two classes. If not provided, the labels default to
+        ``-1`` and ``+1`` (or one-hot encoded equivalents).
+
+    Returns
+    -------
+    Tuple
+        A tuple containing:
+
+        * **training_features** : ``np.ndarray``
+        * **training_labels** : ``np.ndarray``
+        * **testing_features** : ``np.ndarray``
+        * **testing_labels** : ``np.ndarray``
+
+        If ``include_sample_total=True``, a fifth element (``int``) is included
+        that specifies the total number of accepted samples.
+    """
 
-    Returns:
-        Training and testing samples.
+    # Default Value
+    if class_labels is None:
+        class_labels = [0, 1]
+
+    # Errors
+    if training_size < 0:
+        raise ValueError("Training size can't be less than 0")
+    if test_size < 0:
+        raise ValueError("Test size can't be less than 0")
+    if n <= 0:
+        raise ValueError("Number of qubits can't be less than 0")
```

Comment:
```suggestion
        raise ValueError("Number of qubits can't be less than 1")
```
Because you correctly impose `n <= 0`.